br/pkg/restore/snap_client/systable_restore.go

```diff
@@ -275,20 +276,35 @@ func (rc *SnapClient) replaceTemporaryTableToSystable(ctx context.Context, ti *m
 	}
 
 	if db.ExistingTables[tableName] != nil {
-		log.Info("replace into existing table",
+		log.Info("replace existing table",
 			zap.String("table", tableName),
 			zap.Stringer("schema", db.Name))
+
 		// target column order may different with source cluster
 		columnNames := make([]string, 0, len(ti.Columns))
 		for _, col := range ti.Columns {
 			columnNames = append(columnNames, utils.EncloseName(col.Name.L))
 		}
 		colListStr := strings.Join(columnNames, ",")
 		replaceIntoSQL := fmt.Sprintf("REPLACE INTO %s(%s) SELECT %s FROM %s;",
-			utils.EncloseDBAndTable(db.Name.L, tableName),
+			utils.EncloseDBAndTable(db.TemporaryName.L, tableName),
```

Comment:
for this shortcut enhancement, please further check if the system tables in backup and restore cluster have exactly the same schema. otherwise, use the old way.